sound/soc/sof/compress.c

```diff
@@ -10,6 +10,27 @@
 #include "sof-audio.h"
 #include "sof-priv.h"
 
+struct sof_compr_stream {
+	unsigned int copied_total;
```

Comment:
make sure it's a u64, we've had issues before with 32-bit

Reply:
I see. I can have this u64. We have a bit of API misalignment. Here how `snd_compr_tstamp` looks like.

```
struct snd_compr_tstamp {                                                                                                                                                                                  
»       __u32 byte_offset;
»       __u32 copied_total;
»       __u32 pcm_frames;
»       __u32 pcm_io_frames;
»       __u32 sampling_rate;
} __attribute__((packed, aligned(4)));
```

Reply:
sorry, this seems to be still a problem.

The fix I was referring to was 6733cf572a9e20db2b7580a5dd39d5782d571eec ("ALSA: compress: Fix 64bit ABI incompatibility") but I misunderstood the fix. Instead of going to 64 bits the PCM is limited to 32 bits. This isn't good.

This would overflow in 6.21 hours for 192 kHz. Meh.

Reply:
Oh, man. This was a poor decision. But 6 hours of audio should be enough for everyone. :(